### DISCUSSION TITLE
Delete request with query params using the APIClient

### DISCUSSION THREAD

**Discussion by madjaqk** [encode/django-rest-framework#8960] [Question & Answer]
Is there a convenient way when using the `rest_framework.test.APIClient` to make a DELETE request with query parameters?  `my_client.delete("/url/", {"foo": "bar"})` seems to always encode the data in the request body, and I haven't seen any option that would override that.

I know I can add the query params manually, something like `my_client.delete("/url/?" + urllib.parse.urlencode({"foo": "bar"})`, but that's a little awkward; I was hoping this would be possible with just the `APIClient`.

**Comment by cloutierjo** [comment]
Apparently in the next version (5.1?) `query_params` will be usable on any method https://docs.djangoproject.com/en/dev/topics/testing/tools/#django.test.Client.get

but in any previous version, it seems only url encoding is supported for delete and any non GET method in fact
